addons/binding/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/handler/SysteminfoHandler.java

```diff
@@ -81,15 +82,20 @@
      */
     private static final int WAIT_TIME_CHANNEL_ITEM_LINK_INIT = 1;
 
-    private OshiSysteminfo systeminfo;
+    private SysteminfoInterface systeminfo;
 
     ScheduledFuture<?> highPriorityTasks;
     ScheduledFuture<?> mediumPriorityTasks;
 
     private Logger logger = LoggerFactory.getLogger(SysteminfoHandler.class);
 
-    public SysteminfoHandler(Thing thing) {
+    public SysteminfoHandler(@NonNull Thing thing, SysteminfoInterface systeminfo) {
         super(thing);
+        if (systeminfo != null) {
+            this.systeminfo = systeminfo;
+        } else {
+            throw new RuntimeException("No systeminfo service was provided");
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.

Reply:
In this case both an NPE and illegalArgumentExcpetion seem okay